### PR TITLE
Fix mkdirp

### PR DESCRIPTION
**Pull request by willemneal**
- Add fs methods check so that it has default behavior in NEARStudio, since the write file function handles mkdirp

### CODE REVIEW

File: blank_project/package.json [hunk 1]
```diff
@@ -12,8 +12,6 @@
     "test": "npm run build && jest test --env=near-shell/test_environment"
   },
   "devDependencies": {
-    "assemblyscript-json": "github:nearprotocol/assemblyscript-json",
```
Comment (vgrichina) [inline blank_project/package.json]: why are these getting removed?

File: gulp-utils.js [hunk 2]
```diff
@@ -50,17 +56,30 @@ function getAsc() {
...
         const name = filename.startsWith("../") ? filename.substring(3) : filename;
+        if (fs.mkdirSync){
+          mkdirp(name);
+        }
         fs.writeFileSync(name, contents);
       },
-      listFiles: () => []
+      listFiles: (dirname, baseDir) => {
+        if (fs.readdirSync == undefined){
+          return [];
```
Comment (vgrichina) [inline gulp-utils.js]: hmm, if it's ok to remove empty list from `listFiles` and it doesn't break anything – why do we need real implementation at all?

File: gulp-utils.js [hunk 3]
```diff
@@ -31,9 +21,25 @@ function getAsc() {
   }
 
   asc = require("assemblyscript/bin/asc");
-
+  
   const fs = require("fs");
   const pathModule = require("path");
+
+  // Create parent directories if they don't exist
+  function mkdirp(path){
```
Comment (vgrichina) [inline gulp-utils.js]: it doesn't really work like `mkdir -p` (e.g. if you pass `foo/bar.txt` you'd expect `bar.txt` directory created by `mkdir-p`), so either logic or name should be changed

File: gulp-utils.js [hunk 4]
```diff
@@ -1,13 +1,3 @@
-// FUTURE PEOPLE: This file is called "gulp-utils" but it's not related to the deprecated library called "gulp-utils". Don't panic.
```
Comment (vgrichina) [inline gulp-utils.js]: I think `// FUTURE PEOPLE` part should stay

**Comment by janedegtiareva** [comment]
@willemneal what would you like to do with this PR?

**Pull request closed without merging**